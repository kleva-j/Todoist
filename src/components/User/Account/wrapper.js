import { compose, withState, withHandlers } from "recompose";

import { Account } from ".";
import { ComponentFromStreamWrapper } from "../../Reuseables/Wrapper";
import { FirebaseApp } from '../../../services';

const Component = ComponentFromStreamWrapper(Account);

export const AccountWrapper = compose(
  withState("state", "setState", { loading: false }),
  withHandlers({
    handleSubmit: () => () => {},
    changeEmail: () => (email) => {
      FirebaseApp.auth.updateEmail(email)
        .then()
        .catch();
    },
    validateUpdate: () => () => {}
  })
)(Component);
