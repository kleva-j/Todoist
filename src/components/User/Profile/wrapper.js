import { Form, message } from 'antd';
import { compose, withState, withHandlers } from 'recompose';
import { Profile } from '.';
import { ComponentFromStreamWrapper } from '../../Reuseables/Wrapper';
import { getBase64 } from '../../../helpers'

const Component = ComponentFromStreamWrapper(Profile);

export const ProfileWrapper = compose(
  Form.create({ name: "account" }),
  withState("state", "setState", { loading: false, imageUrl: undefined }),
  withHandlers({
    handleSubmit: ({ form }) => (event) => {
      event.preventDefault();
      form.validateFields((err, values) => {
        if (!err) {
          console.log('Received these values ', values);
        } else {
          console.log(err);
        }
      });
    },
    handleFileChange: ({ setState }) => ({ file }) => {
      if (file.status === 'uploading') {
        setState({ loading: true });
        return;
      }
      if (file.status === 'done') {
        getBase64(file.originFileObj, imageUrl =>
          setState({
            imageUrl,
            loading: false,
          }),
        );
      }
    },
    handleBeforeUpload: () => (file) => {
      const isJpgOrPng = file.type === 'image/jpeg' || file.type === 'image/png';
      if (!isJpgOrPng) {
        message.error('You can only upload JPG/PNG file!');
      }
      const isLt2M = file.size / 1024 / 1024 < 2;
      if (!isLt2M) {
        message.error('Image must smaller than 2MB!');
      }
      return isJpgOrPng && isLt2M;
    },
  })
)(Component);
