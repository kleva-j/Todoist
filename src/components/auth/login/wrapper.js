import { Form, message } from "antd";
import { connect } from "react-redux";
import { compose, withState, withHandlers } from "recompose";

import { ComponentFromStreamWrapper } from "../../Reuseables/Wrapper";
import { getMessage } from "../../../constants/alertMessages";
import { isEmpty } from "../../../selectors/user";
import { FirebaseApp } from "../../../services";
import { LoginComponent } from ".";

const Component = ComponentFromStreamWrapper(LoginComponent);

export const Login = compose(
  withState("form_", "setForm", { isLogin: true }),
  withState("state", "setState", {
    alert: {
      type: undefined,
      message: undefined,
      description: undefined,
      isVisible: false,
    },
  }),
  connect(isEmpty),
  Form.create({ name: "login_form" }),
  withHandlers({
    updateState: ({ setState }) => (newState) => {
      setState((state) => ({ ...state, ...newState }));
    },
  }),
  withHandlers({
    handleSubmit: ({ form, updateState }) => (authType) => (event) => {
      event.preventDefault();
      form.validateFields((err, values) => {
        if (!err) {
          return FirebaseApp.loginWithCredentials(authType, values)(
            () =>
              updateState({
                alert: { isVisible: true, ...getMessage("loginSuccess") },
              }),
            ({ code }) =>
              updateState({ alert: { isVisible: true, ...getMessage(code) } })
          );
        } else {
          updateState({
            alert: { isVisible: true, ...getMessage("loginValidation") },
          });
        }
      });
    },
    handleReset: ({ updateState }) => (email, callback) => {
      if (email) {
        FirebaseApp.auth
          .sendPasswordResetEmail(email, {
            url: "http://localhost:3000/dashboard",
          })
          .then(() => {
            updateState({ alert: { isVisible: true, ...getMessage("passwordReset") }});
            callback();
          })
          .catch(({ code }) => {
            updateState({ alert: { isVisible: true, ...getMessage(code) } });
            callback();
          });
      }
    },
    switchForm: ({ setForm, form_: { isLogin } }) => (event) => {
      event.preventDefault();
      setForm({ isLogin: !isLogin });
    },
    loginUser: ({ history }) =>
      FirebaseApp.loginWithSocials(
        (_) => history.push("/dashboard"),
        () => message.error('Network error, Please check your network connection.', 4),
      ),
  })
)(Component);
