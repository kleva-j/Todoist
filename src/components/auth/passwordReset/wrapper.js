import { Form } from 'antd';
import { compose, withHandlers } from 'recompose';

import { EmailPrompt } from './emailPrompt';
import { ComponentFromStreamWrapper } from '../../Reuseables/Wrapper';

export const EmailPromptWrapper = compose(
  Form.create({ name: 'reset_email_prompt'}),
  withHandlers({
    handleFormSubmit: ({ form, handleSubmit, toggleModal }) => (event) => {
      event.preventDefault();
      form.validateFields((err, { email }) => {
        if (err) {
          console.log(err);
          return;
        }
        return handleSubmit(email, () => toggleModal(false));
      });
    },
  }),
)(ComponentFromStreamWrapper(EmailPrompt));
