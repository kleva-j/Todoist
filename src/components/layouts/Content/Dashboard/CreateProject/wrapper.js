import { Form, message } from 'antd';
import { compose, withState, withHandlers } from 'recompose';

import { ProjectModal } from '.';
import { ComponentFromStreamWrapper } from '../../../../Reuseables/Wrapper';

const Component = ComponentFromStreamWrapper(ProjectModal);

export const CreateWrapper = compose(
  Form.create({ name: 'New Project' }),
  withState('state', 'setState', { loading: false }),
  withHandlers({
    handleSubmit: ({ form }) => (event) => {
      event.preventDefault();
      form.validateFields((err, _values) => {
        if (!err) {
          message.success('Values entered are valid');
        } else {
          message.error("Error, Please fix error before submission!");
        }
      });
    },
  })
)(Component);
