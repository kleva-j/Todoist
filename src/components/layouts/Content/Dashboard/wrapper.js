import { compose, withState, withHandlers } from "recompose";
import { Dashboard } from '.';

import { ComponentFromStreamWrapper } from "../../../Reuseables/Wrapper";

const Component = ComponentFromStreamWrapper(Dashboard);

export const DashboardWrapper = compose(
  withState("state", "setState", { loading: false, modalIsVisible: false, modal: {} }),
  withHandlers({
    setModalVisibility: ({ state, setState }) => (value) => {
      setState({ ...state, modalIsVisible: value });
    },
    goToProjects: ({ history }) => () => history.push('/project/all')
  }),
  withHandlers({
    handleOk: ({ setModalVisibility }) => () => setModalVisibility(false),
    handleCancel: ({ setModalVisibility }) => () => setModalVisibility(false),
    toggleCreateModal: ({ setModalVisibility }) => () => setModalVisibility(true),
  }),
)(Component);
