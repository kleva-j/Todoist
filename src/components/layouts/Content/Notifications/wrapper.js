import { Notification } from ".";
import { ComponentFromStreamWrapper } from "../../../Reuseables/Wrapper";
import { compose, withState } from "recompose";

const Component = ComponentFromStreamWrapper(Notification);

export const NotificationsWrapper = compose(
  withState('state', 'setState', {})
)(Component);
