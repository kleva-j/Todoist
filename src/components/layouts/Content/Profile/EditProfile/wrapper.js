import { Form } from 'antd';
import { compose, withState, withHandlers } from 'recompose';
import { EditProfile } from '.';
import { ComponentFromStreamWrapper } from '../../../../Reuseables/Wrapper';

const Component = ComponentFromStreamWrapper(EditProfile);

export const EditWrapper = compose(
  Form.create({ name: 'Edit_Profile' }),
  withState('state', 'setState', {}),
  withHandlers({
    handleSubmit: ({ form }) => (event) => {
      event.preventDefault();
      form.validateFields((err, values) => {
        if (!err) {
          console.log('Received these values ', values);
        }
      });
    }
  })
)(Component);
