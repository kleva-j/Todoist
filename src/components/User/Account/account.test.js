import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { AccountWrapper } from './wrapper';
import { Account } from '.';

configure({ adapter: new Adapter() });

describe('Account container', () => {

  describe('AccountWrapper', () => {
    const wrapper = shallow(<AccountWrapper />);
    const props = wrapper.props();
  
    test('should render correctly', () => {
      expect(props).toHaveProperty('state');
      const prop = wrapper.prop('state');
      expect(prop).toHaveProperty('loading', false);
    });
    
    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Account component', () => {
    const wrapper = shallow(<Account />);
    
    test('should render correctly', () => {
      const prop = wrapper.prop('children');
      expect(prop.length).toBe(2);
      expect(prop[0].props.children.length).toBe(4);
      expect(prop[1].props.children.length).toBe(4);
    });
    
    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
  });
});
