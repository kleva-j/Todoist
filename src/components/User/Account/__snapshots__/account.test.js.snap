// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account container Account component should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Account container AccountWrapper should match snapshot 1`] = `ShallowWrapper {}`;
